src/InsightContainer/index.js

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { makeStyles } from '@material-ui/styles';
+import Grid from '@material-ui/core/Grid';
+import { Typography } from '@material-ui/core';
+import BlockLoader from '../BlockLoader';
+import TypographyLoader from '../TypographyLoader';
+
+import A from '../A';
+import InsightDataActions from './InsightDataActions';
+
+const useStyles = makeStyles({
+  root: {
+    width: 'available',
+    height: 'auto',
+    backgroundColor: '#fff'
+  },
+  content: {
+    padding: '1.25rem 0',
+    overflow: 'hidden'
+  },
+  button: {
+    border: '0.0625rem solid #d8d8d8',
+    height: '2.5rem',
+    width: '2.5rem',
+    '&:first-child': {
+      borderRight: 'none'
+    }
+  },
+  actionsGrid: {
+    backgroundColor: '#eeebeb'
+  },
+  title: {
+    marginTop: '1rem',
+    fontWeight: 'bold',
+    fontSize: '1.25rem'
+  },
+  subtitle: {},
+  sourceLink: {},
+  analysisLink: {
+    cursor: 'pointer',
+    display: 'inline-flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    width: '100%',
+    padding: '1.125rem 0',
+    textDecoration: 'none',
+    outline: 'none',
+    borderRadius: '0.75rem',
+    color: '#29a87c',
+    border: 'solid 2px #29a87c',
+    backgroundColor: '#fff'
+  },
+  summaryGrid: {
+    padding: '0 1.25rem'
+  },
+  summaryHead: {
+    fontWeight: 'bold',
+    fontSize: '1rem'
+  },
+  summaryBrief: {
+    fontSize: '0.8125rem',
+    lineHeight: 2
+  }
+});
+
+function InsightContainer({
+  loading,
+  content,
+  sourceUrl,
+  title,
+  children,
+  analysis,
+  analysisUrl,
+  summaryBrief,
+  summaryHead
+}) {
+  const classes = useStyles();
+
+  const handleShare = () => {};
+
+  const handleDownload = () => {};
+
+  const handleShowData = () => {};
+
+  const handleComapre = () => {};
```

Comment:
lets have these from a prop
```
chartActions: {
  handleShare: () => {},
  handleDownload: () => {},
  handleShowData: () => {},
  handleComapre: () => {},
}
```
or 
```
dataActions: {
  handleShare: () => {},
  handleDownload: () => {},
  handleShowData: () => {},
  handleComapre: () => {},
}
```
since the component is called insightdataactions

the default actions can be defined and if someone passes their own function then they can handle their own behaviour.